lib/spack/llnl/util/filesystem.py

```diff
@@ -2278,10 +2278,17 @@ def add_rpath(self, *paths):
         """
         self._addl_rpaths = self._addl_rpaths | set(paths)
 
-    def _link(self, path, dest):
+    def _link(self, path, dest_dir):
+        """Perform link step of simulated rpathing, installing
+        simlinks of file in path to the dest_dir
+        location. This method deliberately prevents
+        the case where a path points to a file inside the dest_dir.
+        This is because it is both meaningless from an rpath
+        perspective, and will cause an error when Developer
+        mode is not enabled"""
```

Comment:
Reading the code it seems the function now just returns if we try to link on the same location. I'm not clear though on what the docstring is saying... Can you help me clarifying how the docstring relates to the change?

Reply:
the docstring is written from a particular use case, it doesn't belong there

Reply:
This is a private method used only once inside of a class with a specific use case. Normally I'd agree about where this type of check should occur, but IMO private methods have more leeway. If you disagree, would you want to add a TODO?